public/app/features/templating/state/selectors.ts

```diff
@@ -0,0 +1,32 @@
+import { cloneDeep } from 'lodash';
+
+import { StoreState } from '../../../types';
+import { VariableModel } from '../variable';
+import { getState } from '../../../store/store';
+import { EMPTY_UUID } from './types';
+
+export const getVariable = <T extends VariableModel = VariableModel>(
+  uuid: string,
+  state: StoreState = getState()
+): T => {
+  if (!state.templating.variables[uuid]) {
+    throw new Error(`Couldn't find variable with uuid:${uuid}`);
+  }
+
+  return state.templating.variables[uuid] as T;
+};
+
+export const getVariableWithName = (name: string, state: StoreState = getState()) => {
+  return Object.values(state.templating.variables).find(variable => variable.name === name);
```

Comment:
wish this was an index lookup as it's called 10-100 times per refresh (depends on how many panels and queries), but don't think it's critical

Reply:
Yes, I've a branch that removes uuid from state. Will work on that after this is merged to master

Reply:
Added to #22564